imported and not used: "strings"
make: *** [all] Error 2

Also 'labels' map var could be removed then.

And one more issue:
needs fix then ploop-flexvol to proper build vzsReplicas(1.2.3 -> 1:2/3) and vzsEncoding(1.2.3 -> 1+2/3) values.

BTW:
not sure, but may be place any 'vstorage' calls out of ploop-flexvol. If we can use ploop-flexvol for any underlying storage, why do we adapt it for vstorage only ?
I see it's more correct to place code for calling 'vstorage' in virtuozzo-provisioner main package, rather then ploop-flexvol. Am I right ?

--- a/vzstorage-pd/README.md
+++ b/vzstorage-pd/README.md
@@ -47,26 +47,5 @@ kubectl create -f test-pod.yaml
 
 A storage class parameters pass as ploop options to the ploop-flexvol driver.
 
-Another way to pass ploop options is to use labels in claims. In this case all special symbols (e.g. +,/, etc) have to be replaced by dots.
-
-Labels in claims can be used to pass ploop options, but all special symbols has to be replaced by dots.
- kind: PersistentVolumeClaim
- apiVersion: v1
- metadata:
-   name: vz-test-claim
-   annotations:
-     volume.beta.kubernetes.io/storage-class: "virtuozzo-storage"
- spec:
-   accessModes:
-     - ReadWriteOnce
-     - ReadOnlyMany
-   resources:
-     requests:
-       storage: 1Gi
-   selector:
-     matchLabels:
-       vzsTier: "0"
-       vzsReplicas: "2.1.3"
-
 # Known limitations
 Vstorage must be mounted manually on all cluster nodes

--- a/vzstorage-pd/claim.yaml
+++ b/vzstorage-pd/claim.yaml
@@ -11,7 +11,3 @@ spec:
   resources:
     requests:
       storage: 1Gi
-  selector:
-    matchLabels:
-      vzsTier: "0"
-      vzsReplicas: "2.1.3"

--- a/vzstorage-pd/virtuozzo-provisioner.go
+++ b/vzstorage-pd/virtuozzo-provisioner.go
@@ -21,7 +21,6 @@ import (
 	"flag"
 	"fmt"
 	"os"
-	"strings"
 	"time"
 
 	"github.com/golang/glog"
@@ -72,22 +71,17 @@ var _ controller.Provisioner = &vzFSProvisioner{}
 func (p *vzFSProvisioner) Provision(options controller.VolumeOptions) (*v1.PersistentVolume, error) {
 	var (
 		capacity resource.Quantity
-		labels   map[string]string
 	)
 
 	capacity = options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
 	bytes := capacity.Value()
 
-	if options.PVC.Spec.Selector != nil && options.PVC.Spec.Selector.MatchExpressions != nil {
-		return nil, fmt.Errorf("claim Selector.matchExpressions is not supported")
+	if options.PVC.Spec.Selector != nil {
+		return nil, fmt.Errorf("claim Selector is not supported")
 	}
 	share := fmt.Sprintf("kubernetes-dynamic-pvc-%s", uuid.NewUUID())
 
 	glog.Infof("Add %s %s", share, capacity.Value())
-
-	if options.PVC.Spec.Selector != nil && options.PVC.Spec.Selector.MatchLabels != nil {
-		labels = options.PVC.Spec.Selector.MatchLabels
-	}
 
 	ploop_options := map[string]string{}
 	for k,v := range options.Parameters {
@@ -96,27 +90,6 @@ func (p *vzFSProvisioner) Provision(options controller.VolumeOptions) (*v1.Persi
 
 	ploop_options["volumeId"] = share
 	ploop_options["size"] = fmt.Sprintf("%d", bytes)
-
-	if labels != nil {
-		for k, v := range labels {
-			switch k {
-			case "vzsReplicas":
-				v = strings.Replace(v, ".", ":", 1)
-				v = strings.Replace(v, ".", "/", 1)
-				ploop_options[k] = v
-			case "vzsTier":
-				fallthrough
-			case "vzsEncoding":
-				v = strings.Replace(v, ".", "+", 1)
-				v = strings.Replace(v, ".", "/", 1)
-				ploop_options[k] = v
-			case "vzsFailureDomain":
-				ploop_options[k] = v
-			default:
-				glog.Infof("Skip %s = %s", k, v)
-			}
-		}
-	}
 
 	if err := volume.Create(ploop_options); err != nil {
 		return nil, err
@@ -129,7 +102,6 @@ func (p *vzFSProvisioner) Provision(options controller.VolumeOptions) (*v1.Persi
 				provisionerIDAnn: string(p.identity),
 				vzShareAnn:       share,
 			},
-			Labels: labels,
 		},
 		Spec: v1.PersistentVolumeSpec{
 			PersistentVolumeReclaimPolicy: options.PersistentVolumeReclaimPolicy,

